hermes_eea/SkymapFactory.py

```diff
@@ -0,0 +1,102 @@
+
+import numpy as np
+from hermes_core import log
+from hermes_eea.io import EEA
+from hermes_eea.util.time import ccsds_to_cdf_time
+from hermes_eea import energies as voltages
+
+# This may eventually be put inside a multiprocessor:
+def SkymapFactory(l0_cdf,energies,deflections,myEEA):
+    #['Epoch', 'Epoch_plus_var', 'Epoch_minus_var', 'hermes_eea_step_counter',
+    #'hermes_eea_counter1', 'hermes_eea_counter2', 'hermes_eea_accumulations',
+    #'hermes_eea_sector_index', 'hermes_eea_sector_label'])
+
+    # science_data:
+    start_of_good_data = np.where(l0_cdf['SHEID'][:] == 1)[0][0]
+    integrates_at_end = np.where(l0_cdf['SHEID'][start_of_good_data:] == 0) # has 63 values
+    # We are expecting integrates to be only at the beginning
+
+    # The Science Data:
+    stepper_table_packets = (np.where(l0_cdf['SHEID'][:] > 0))[0]
+    return_package = {}
+    beginning_packets = np.where((l0_cdf['STEP'][stepper_table_packets[0]:]) == 0 )[0] + stepper_table_packets[0]
+    package = []
+
+    epochs = ccsds_to_cdf_time.helpConvertEEA(l0_cdf)
+    try:
+        for ptr in range(0,len(beginning_packets)):
+            package.append((
+                l0_cdf['STEP'][beginning_packets[ptr]:beginning_packets[ptr+1]],
+                l0_cdf['ACCUM'][beginning_packets[ptr]:beginning_packets[ptr+1]],
+                l0_cdf['COUNTER1'][beginning_packets[ptr]:beginning_packets[ptr+1]],
+                l0_cdf['COUNTER2'][beginning_packets[ptr]:beginning_packets[ptr+1]],
+                epochs[beginning_packets[ptr]:beginning_packets[ptr+1]],
+
+                energies, deflections,ptr
+            ))
+    except IndexError:
+        log.info("Finished last interval")
+
+    result = []
+    for pckt in package:
+                packet_contents = do_eea_packet(*pckt)
+                if packet_contents != None:
+                   result.append(packet_contents)
+    myEEA.populate(myEEA, result)
+
+    #epoch = ccsds_to_cdf_time.helpConvert_eea(l0_cdf)
+    #zero_values_past_first = np.where(l0_cdf['hermes_eea_intgr_or_stepper'][135:] == 0)[0]
+    #l0_cdf['hermes_eea_step_counter'][zero_values_past_first]
+    #first_packages = np.where(l0_cdf['SHEID'] > 0)
+
+# This does an entire sweep of, nominally, 164 thingies
+def do_eea_packet( stepperTableCounter,
+                   counts,
+                   cnt1,cnt2,
+                   epoch,
+                   energies,
+                   deflections,ith_FSmap):
+
+     return_package = {}
+     rows = len(stepperTableCounter)
+     # skymap is already full of zeros, why do it again?
+     # skymap = np.zeros((beginning_packets[ptr+1]-beginning_packets[ptr],32))
+     skymaps = []
+     pulse_a = np.zeros((41,4), dtype=np.uint16)
```

Comment:
Use a global variable that describes these dimensions and put them at the top of the file.